draft-plugin/server.go

```diff
@@ -0,0 +1,200 @@
+		},
+	)
+	_ = l
+	var c *config.Config
+	var p = getProwClient()
+	c, err := config.Load(s.prowconfig, s.prowjob, nil, "")
+	if err != nil {
+		logrus.Info(err)
+	}
+	var pr github.PullRequestEvent
+	if err := json.Unmarshal(payload, &pr); err != nil {
+		return err
+	}
+	org, repo, author, ref := orgRepoAuthor(pr.PullRequest)
+	baseSHA := ""
+	baseSHAGetter := func() (string, error) {
+		var err error
+		baseSHA, err = githubClient.GetRef(s.ghc, org, repo, "heads/"+ref)
+		if err != nil {
+			return "", fmt.Errorf("failed to get baseSHA: %w", err)
+		}
+		return baseSHA, nil
+	}
+	headSHAGetter := func() (string, error) {
+		return pr.PullRequest.Head.SHA, nil
+	}
+	presubmits, err := c.GetPresubmits(s.gc, org+"/"+repo, baseSHAGetter, headSHAGetter)
+	if err != nil {
+		fmt.Printf("server not responding %s", err.Error())
+	}
+	if len(presubmits) == 0 {
+		return nil
+	}
+	trustedResponse, err := trigger.TrustedUser(s.ghc, true, pr.Repo.Owner.Login, author, org, repo)
+	member := trustedResponse.IsTrusted
+	if err != nil {
+		return fmt.Errorf("could not check membership: %s", err)
+	}
+	if member {
+		switch pr.Action {
+		case github.PullRequestActionOpened, github.PullRequestActionSynchronize, github.PullRequestActionConvertedToDraft:
+			infomation := "{\"action\": \"" + string(pr.Action) + "\",\"author\": \"" + author + "\",\"pr\": " + strconv.Itoa(pr.Number) + "\",\"repo\": \"" + pr.Repo.Name + "\",\"SHA\": \"" + *pr.PullRequest.MergeSHA + "\"}"
+			logrus.Info(infomation)
+			if pr.PullRequest.Draft {
+				var jobs []config.Presubmit
+				for _, job := range presubmits {
+					match, err := regexp.MatchString(s.regex, job.Name)
+					if err != nil || !match {
+						continue
+					}
+					jobs = append(jobs, job)
+				}
+				var errors []error
+				for _, job := range jobs {
+					pjob := map[string][]config.Presubmit{
+						job.Name: {job},
+					}
+					err := c.JobConfig.SetPresubmits(pjob)
+					if err != nil {
+						logrus.Error(err)
+					}
+					logrus.Infof("Starting %s build.", job.Name)
+					pj := pjutil.NewPresubmit(pr.PullRequest, baseSHA, job, eventGUID, nil)
+					logrus.WithFields(pjutil.ProwJobFields(&pj)).Info("Creating a new prowjob.")
+					if err := createWithRetry(context.Background(), p.ProwJobs("default"), &pj, time.Millisecond); err != nil {
+						logrus.WithError(err).Error("Failed to create prowjob.")
+						errors = append(errors, err)
+					}
+				}
+				logrus.Info(errors)
+			}
+		default:
+			logrus.Debugf("skipping event of type %q", eventType)
+		}
+	}
+	return nil
+}
+
+func orgRepoAuthor(pr github.PullRequest) (string, string, string, string) {
+	org := pr.Base.Repo.Owner.Login
+	repo := pr.Base.Repo.Name
+	author := pr.User.Login
+	ref := pr.Base.Ref
+	return org, repo, author, ref
+}
+
+// createWithRetry will retry the cration of a ProwJob. The Name must be set, otherwise we might end up creating it multiple times
+// if one Create request errors but succeeds under the hood.
```

Comment:
```suggestion
// createWithRetry will retry the creation of a ProwJob. The Name must be set, otherwise we might end up creating it multiple times
// if one Create request throws error but succeeds under the hood.
```